Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -334,55 +315,29 @@ public override List<Cash> GetCashBalance()
         /// Gets the execution details matching the filter
         /// </summary>
         /// <returns>A list of executions matching the filter</returns>
-        public List<IB.ExecDetailsEventArgs> GetExecutions(string symbol, IB.SecurityType? type, string exchange, DateTime? timeSince, IB.ActionSide? side)
+        public ConcurrentDictionary<int, ExecutionDetails> GetExecutions(string symbol, string type, string exchange, DateTime timeSince, string side)
         {
-            var filter = new IB.ExecutionFilter
+            
+            var filter = new ExecutionFilter()
             {
                 AcctCode = _account,
                 ClientId = _clientID,
                 Exchange = exchange,
-                SecurityType = type ?? IB.SecurityType.Undefined,
+                SecType = type ?? "",
                 Symbol = symbol,
-                Time = timeSince ?? DateTime.MinValue,
-                Side = side ?? IB.ActionSide.Undefined
+                Time = timeSince.ToString("yyyymmdd hh:mm:ss") ?? DateTime.MinValue.ToString("yyyymmdd hh:mm:ss"),
+                Side = side ?? ""
             };
-
-            var details = new List<IB.ExecDetailsEventArgs>();
-            using (var client = new IB.IBClient())
+            var client = new EClientSocket(this);
+            client.eConnect(_host, _port, IncrementClientID());
```

Comment:
Makes sense. This one is almost exactly like the old implementation.
I will change the code according to this new pattern.
it looks the same as the old one but with newer callbacks. Events remain the same. 
thanx for writing up the event args classes. 
I'll get to it.